public/coffee/directives/asyncForm.coffee

```diff
@@ -51,14 +52,15 @@ define [
 								else
 									ga('send', 'event', formName, 'success')
 
-						.error (data, status, headers, config) ->
+						.catch (httpResponse) ->
+							{ data, status, headers, config } = httpResponse
 							scope[attrs.name].inflight = false
 							response.success = false
 							response.error = true
 
 							onErrorHandler = scope[attrs.onError]
 							if onErrorHandler
-								onErrorHandler(data, status, headers, config)
+								onErrorHandler(httpResponse)
```

Comment:
Just a note for anyone reviewing, but the only place these `onErrorHandler` and `onSuccessHandler` are used, is in the `launchpad` module.